From 6df976eb6a078ea81b32c218b8c4fec1faf7c02e Mon Sep 17 00:00:00 2001
From: Jonathon Jongsma <jjongsma@redhat.com>
Date: Tue, 2 Sep 2014 16:45:00 -0500
Subject: [PATCH] Set initial window size to display desktop size

When setting the 'display' for a VirtViewerWindow, the initial size for
that window should be the size of the remote display. So we synthesize a
desktop resize event when setting a new display for a window.  This is
only done for enabled displays. Disabled displays generally have a size
of 0x0, which would result in the window being at it's minimum size, so
just allow the window to use its default size.

(cherry picked from commit 29c8cadd4037681e8eb417fd8a0bb6cc6170a6b3)
---
 src/virt-viewer-window.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/virt-viewer-window.c b/src/virt-viewer-window.c
index 8dbf5a5..7182c2f 100644
--- a/src/virt-viewer-window.c
+++ b/src/virt-viewer-window.c
@@ -1327,6 +1327,8 @@ virt_viewer_window_set_display(VirtViewerWindow *self, VirtViewerDisplay *displa
                                           G_CALLBACK(virt_viewer_window_desktop_resize), self, 0);
         virt_viewer_signal_connect_object(display, "notify::show-hint",
                                           G_CALLBACK(display_show_hint), self, 0);
+        if (virt_viewer_display_get_enabled(display))
+            virt_viewer_window_desktop_resize(display, self);
     }
 }
 
